executor/cte.go

```diff
@@ -248,6 +265,9 @@ func (e *CTEExec) computeRecursivePart(ctx context.Context) (err error) {
 			if err = e.setupTblsForNewIteration(); err != nil {
 				return err
 			}
+			if e.hasLimit && uint64(e.resTbl.NumRows()) >= e.limitEnd {
```

Comment:
Better use a function to wrap e.hasLimit && uint64(e.resTbl.NumRows()) >= e.limitEnd

Reply:
Done. use `limitDone()` function to check.